### PR TITLE
Pm 1542 modified delete APIs to have a 409 HTTP status in response when the payment is in DENIED status

### PR DESCRIPTION
**Pull request by albertogelmi-sia**
#### List of Changes

modified delete APIs to have a 409 HTTP status in response when the payment is in DENIED status

#### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [X] My change requires a change to the documentation.
- [X] I have updated the documentation accordingly.
- [X] I have created or updated an API interface.
  - [X] I have updated the OpenAPI spec
- [ ] I have created or updated a database table
  - [ ] I have updated both the Oracle and Postgres migrations
- [X] My PR is cohesive and self-contained (rule of thumb: less than 250 changed lines) 

### Link to story

https://pagopa.atlassian.net/browse/PM-1542

### CODE REVIEW

**Comment by giovanniberti** [comment]
Please add tests for the DENIED + refund case :)

**Comment by robertorasi-nexigroup** [comment]
> Please add tests for the DENIED + refund case :)

Added new tests for vpos and xpay

File: src/main/java/it/pagopa/pm/gateway/service/VposDeleteService.java [hunk 1]
```diff
@@ -65,6 +64,11 @@ public VposDeleteResponse startDelete(String requestId) {
             return createDeleteResponse(requestId, REQUEST_ID_NOT_FOUND_MSG, null);
         }
 
+        if (DENIED.name().equals(entity.getStatus())) {
+            log.info("Payment with requestId " + requestId + " is in DENIED status. Skipping refund");
```
Comment (giovanniberti) [inline src/main/java/it/pagopa/pm/gateway/service/VposDeleteService.java:68]:
```suggestion
            log.info("Payment with requestId {} is in DENIED status. Skipping refund", requestId);
```

File: src/test/java/it/pagopa/pm/gateway/controller/CreditCardPaymentControllerTest.java [hunk 2]
```diff
@@ -179,4 +179,16 @@ public void deleteVposPayment_Test_500() throws Exception {
                 .andExpect(status().isInternalServerError());
     }
 
+    @Test
+    public void deleteVposPayment_Test_409() throws Exception {
+        String UUID_SAMPLE = "8d8b30e3-de52-4f1c-a71c-9905a8043dac";
```
Comment (giovanniberti) [inline src/test/java/it/pagopa/pm/gateway/controller/CreditCardPaymentControllerTest.java:184]:
```suggestion
        String transactionId = UUID.randomUUID().toString();
```

**Review by giovanniberti — APPROVED** [review]

**Pull request merged**